Assignment2/Logger.Tests/BaseLoggerMixinsTests.cs

```diff
@@ -0,0 +1,99 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections.Generic;
+
+namespace Logger.Tests
+{
+    [TestClass]
+    public class BaseLoggerMixinsTests
+    {
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentException))]
+        public void Error_WithNullLogger_ThrowsException()
+        {
+            // Arrange
+
+            // Act
+            BaseLoggerMixins.Error(null, "");
+
+            // Assert
+        }
+
+        [TestMethod]
+        public void Error_WithData_LogsMessage()
+        {
+            // Arrange
+            var logger = new TestLogger();
+
+            // Act
+            logger.Error("Message {0}", 42);
+
+            // Assert
+            Assert.AreEqual(1, logger.LoggedMessages.Count);
+            Assert.AreEqual(LogLevel.Error, logger.LoggedMessages[0].LogLevel);
+            Assert.AreEqual("Message 42", logger.LoggedMessages[0].Message);
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentException))]
+        public void WarningWithNullLogger()
```

Comment:
Use descriptive names for your automated tests. As a best practice, they should follow the pattern of 'what is being tested', 'what is the test case', and 'what is the expected outcome':
```suggestion
        public void LoggerWarning_WithNullLogger_ThrowsException()
```